### PR TITLE
docs: [IAC-2473] IaCM workspace templates

### PR DESCRIPTION
**Pull request by richardblack-Harness**
## Description

* IaCM workspace templates guide
* [JIRA](https://harness.atlassian.net/browse/IAC-2473)

### Screenshot / Preview
https://github.com/user-attachments/assets/750429aa-9698-4a46-a719-6b9577dd74d5

PRs must meet these requirements to be merged:
- [x] Successful preview build.
- [x] Code owner review.
- [x] No merge conflicts.
- [x] Release notes/new features docs: Feature/version released to at least one prod environment.

### CODE REVIEW

**Comment by richardblack-Harness** [comment]
Interactive guide links:

- [Create a workspace template](https://app.tango.us/app/workflow/Creating-an-IaCM-workspace-template-in-Harness-6686c1eafb9b4a94ad1e021c34b8a19f/play)
- [Add variables and review template](https://app.tango.us/app/workflow/Adding-variables-to-your-Harness-IaCM-workspace-template-ceb4cbaeaae840d58b6870e72cd4a7cf/play)
- [Apply workspace templates to a new workspace](https://app.tango.us/app/workflow/Apply-Harness-IaCM-workspace-templates-to-your-new-workspace--fb4819c2398548da90f41db8ae839fc5/play)

**Review by crystal-fong — COMMENTED** [review]
looks good! just some minor comments

File: docs/infra-as-code-management/workspaces/workspace-tempates.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+   3. To edit, select **Open in Template Studio**.
+   4. Update your workspace settings and select **Save**.
+
+   ## Apply your Workspace Template
+   1. In one of your projects, create a new workspace.  
+   2. When prompted, choose one of the following options:  
+      - **Start from Scratch** to create a workspace without a template.  
+      - **Start with Template** to apply a pre-configured template.  
+   3. Select **Start with Template** and assign a name to the new workspace.  
+   4. Review the applied configuration and variables.
```
[the remark this answers is not in the document]
Comment (richardblack-Harness) [reply docs/infra-as-code-management/workspaces/workspace-tempates.md:111]: Probably, `Use Template` seemed like the natural option, but it was disabled.

Remind me, what was the difference again?

File: docs/infra-as-code-management/workspaces/workspace-tempates.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+/>
+
+import Tabs from '@theme/Tabs';  
+import TabItem from '@theme/TabItem';  
+ 
+With workspace templates, you can standardize workspace configurations across your projects by predefining essential variables, configuration settings, and other workspace options. This helps streamline workspace setup, ensuring consistency and reducing manual configuration efforts.  
+
+### Prerequisites
+Before you begin configuring workspace templates, ensure that you have:
+- Admin or template creation permissions.
```
[the remark this answers is not in the document]
Comment (richardblack-Harness) [reply docs/infra-as-code-management/workspaces/workspace-tempates.md:24]: No, that's a fair point. Done

File: docs/infra-as-code-management/workspaces/workspace-tempates.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (crystal-fong) [reply docs/infra-as-code-management/workspaces/workspace-tempates.md:111]: so once a user "uses a template", their workspace will follow any changes made to the template. Copying a template is a one-off action

[the remark this answers is not in the document]
Comment (richardblack-Harness) [reply docs/infra-as-code-management/workspaces/workspace-tempates.md]: Yes, that rings a bell now, thanks for clarifying. I've added this to a few place, it's an important point.

**Review by crystal-fong — APPROVED** [review]

**Review by RickCraig — APPROVED** [review]

**Pull request merged**